atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -495,6 +514,19 @@ private TransactionManager serializableInternal(@Output List<AutoCloseable> clos
         return transactionManager;
     }
 
+    private TimestampCache getTimestampCache(
+            MetricsManager metricsManager,
+            Supplier<AtlasDbRuntimeConfig> runtimeConfigSupplier,
+            Optional<PersistentTimestampStore> persistentTimestampStore) {
+        LongSupplier cacheSize = () -> runtimeConfigSupplier.get().getTimestampCacheSize();
+        Supplier<TimestampCache> timestampCacheSupplier = () ->
+                persistentTimestampStore.map(store ->
+                        OffHeapTimestampCache.create(store, metricsManager.getTaggedRegistry(), cacheSize))
+                        .orElseGet(() -> new DefaultTimestampCache(metricsManager.getRegistry(), cacheSize));
+
+        return config().timestampCache().orElseGet(timestampCacheSupplier);
```

Comment:
I think we should test the logic in this class in this PR, especially since it's on a live path